### ISSUE TITLE
Should `transaction_id` be set, even if there is no association being saved?

### ISSUE THREAD

**Issue by aried3r** [airblade/paper_trail#606]
I'm not certain if this is a bug or not, but I saw that a `transaction_id` is generated and saved to the model, even if no associations are being saved (using `--with-associations` during install time).

Shouldn't it work without, since you can retroactively add the association migration?

My use case is not directly related to this, I am migrating from a different auditing system and want to manually create versions, using
```Ruby
widget.versions.create(event: 'create', whodunnit: user_id_from_old_auditing_system)
```
and saw it was missing the `transaction_id`, which the ones after adding paper_trail were not. So, either a) it is a "bug" (missing enhancement, really) in paper_trail, that it's setting the `transaction_id` even though it could omit it, b) it's a bug on my side not setting it, c) I shouldn't be manually creating versions :D

`.touch_with_version` does not set a `transaction_id` btw.

Edit: I do not want to `.touch` my entries, otherwise I'd have used `.touch_with_version` to create one.

**Comment by batter** [comment]
I don't believe it's a bug per say, since it doesn't really cause any issues, but I believe you are correct in your assertion that:
> a) paper_trail, that it's setting the `transaction_id` even though it could omit it

I just looked at the source and after reading for a bit it appears as though it [sets the version id during creation, updates, etc](https://github.com/airblade/paper_trail/blob/v4.0.0/lib/paper_trail/has_paper_trail.rb#L307) in preparation for the possibility that it may need to version / link up associations after the fact, if the `transaction_id` column is present.  So maybe we could do some extra work and force it not to insert into that column unless necessary but I wouldn't worry about it for now.

What do you think @jaredbeck?  The association versioning seems brittle enough that I don't want to mess with it too much.

**Comment by aried3r** [comment]
So, to clarify, it's probably not needed. I'm asking because I want to start off without `--with-associations` and probably even without `--with-changes` but want to add these later on as I go, when I need them. But since I've seen `transaction_id` being set, even if there are no associations, I wasn't sure if that will work if I don't have that column from the start.

TL;DR: Can I activate association and change tracking later on?

**Comment by jaredbeck** [comment]
> TL;DR: Can I activate association and change tracking later on?

Yes.

> .. maybe we could do some extra work and force it not to insert into [the transaction_id] column unless necessary but I wouldn't worry about it for now.

I agree.  Given his later comment, I don't think that's what Anton is asking for, anyway.

**Comment by aried3r** [comment]
> Given his later comment, I don't think that's what Anton is asking for, anyway.

Well, yeah. I was looking for a solution for my special use case, but also found it might be a missing 'optimization', which is why I reported it.

Thanks for your feedback!
